src/Schema/AbstractSchemaManager.php

```diff
@@ -521,14 +507,56 @@ protected function selectForeignKeyColumns(string $databaseName, ?string $tableN
     }
 
     /**
-     * Returns table options for the tables in the specified database. If the table name is specified, narrows down
-     * the selection to this table.
+     * Fetches definitions of table columns in the specified database and returns them grouped by table name.
+     *
+     * @return array<string,list<array<string,mixed>>>
+     *
+     * @throws Exception
+     */
+    protected function fetchTableColumnsByTable(string $databaseName): array
```

Comment:
Should some of these new methods be `final`? They don't look like they are intended to be overridden.

Reply:
The point of having these methods extracted is being able to do an override for SQLite. As the next step, I want to introduce a separate API for schema introspection, so we're likely to deprecate these methods in the future.